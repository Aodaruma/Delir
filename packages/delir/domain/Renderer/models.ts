import * as Delir from '@ragg/delir-core'

export const hasErrorInClip = (clip: Delir.Entity.Clip, error: Delir.Exceptions.UserCodeException | null) => {
    if (!error) return false
    if (error.location.type === 'clip' && error.location.entityId === clip.id) return true
    return error.location.type === 'effect' && !!clip.effects.find(effect => error.location.entityId === effect.id)
}
