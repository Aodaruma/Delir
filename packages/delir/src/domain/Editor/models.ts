export const NotificationTimeouts = {
  verbose: 2000,
  error: 6000,
}
