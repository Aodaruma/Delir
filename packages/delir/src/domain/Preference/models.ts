import Joi from 'joi'

const preferenceSchema = Joi.object()
  .keys({
    editor: Joi.object()
      .keys({
        audioVolume: Joi.number()
          .min(0)
          .max(100)
          .required(),
      })
      .strict(),
    develop: Joi.object()
      .keys({
        pluginDirs: Joi.array().items(Joi.string()),
      })
      .strict(),
    renderer: Joi.object()
      .keys({
        ignoreMissingEffect: Joi.boolean().required(),
      })
      .strict(),
  })
  .strict()

export const validateSchema = (obj: any): Joi.ValidationError | null => {
  const result = Joi.validate(obj, preferenceSchema)
  return result.error
}
