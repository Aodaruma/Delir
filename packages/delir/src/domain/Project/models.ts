import * as Delir from '@delirvfx/core'
import immer from 'immer'

const effectReplaceMap: { [oldName: string]: string } = {
  '@ragg/delir-posteffect-color-slider': '@delirvfx/posteffect-color-slider',
  '@ragg/delir-posteffect-numeric-slider': '@delirvfx/posteffect-numeric-slider',
  '@ragg/delir-posteffect-the-world': '@delirvfx/posteffect-the-world',
}

export const migrateProject = (project: Delir.Entity.Project): Delir.Entity.Project => {
  return immer(project, project => {
    Delir.MigrationHelper.walkEffects((project as any) as Delir.Entity.Project, effect => {
      const { processor } = effect
      const newId = effectReplaceMap[processor]

      if (newId) {
        effect.processor = newId
      }
    })
  })
}
