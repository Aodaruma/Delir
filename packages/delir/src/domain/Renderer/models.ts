import * as Delir from '@delirvfx/core'

export const hasErrorInClip = (clip: Delir.Entity.Clip, error: Delir.Exceptions.UserCodeException | null) => {
  if (!error) return false
  if (error.location.type === 'clip' && error.location.entityId === clip.id) return true
  return error.location.type === 'effect' && !!clip.effects.find(effect => error.location.entityId === effect.id)
}
