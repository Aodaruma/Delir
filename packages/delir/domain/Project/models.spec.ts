import { mockClip, mockComposition, mockEffect, mockLayer, mockProject } from '@delirvfx/core-test-helper'
import { times } from 'lodash'
import { migrateProject } from './models'

describe('Project/models', () => {
  it('Should migrate from old effect id to next id', () => {
    const project = mockProject()
    const comp = mockComposition({ id: 'comp1', layers: [mockLayer({ id: ' layerId' })] })
    project.addComposition(comp)

    times(2).forEach(idx => {
      const clip = mockClip({ id: 'clip-${idx}' })
      ;[
        '@ragg/delir-posteffect-color-slider',
        '@ragg/delir-posteffect-numeric-slider',
        '@ragg/delir-posteffect-the-world',
      ].forEach((processor, idx) => {
        clip.addEffect(mockEffect({ id: `layer${idx}`, processor }))
      })

      comp.layers[0].addClip(clip)
    })

    const newProject = migrateProject(project)
    expect(newProject).toMatchSnapshot()
  })
})
