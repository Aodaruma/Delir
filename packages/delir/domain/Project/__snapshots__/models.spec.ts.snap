// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project/models Should migrate from old effect id to next id 1`] = `
Project {
  "assets": Array [],
  "compositions": Array [
    Composition {
      "audioChannels": 2,
      "backgroundColor": Object {
        "blue": 0,
        "green": 0,
        "red": 0,
      },
      "durationFrames": 100,
      "framerate": 30,
      "height": 100,
      "id": "comp1",
      "layers": Array [
        Layer {
          "clips": Array [
            Clip {
              "durationFrames": 0,
              "effects": Array [
                Effect {
                  "expressions": Object {},
                  "id": "layer0",
                  "keyframes": Object {},
                  "processor": "@delirvfx/posteffect-color-slider",
                  "referenceName": null,
                },
                Effect {
                  "expressions": Object {},
                  "id": "layer1",
                  "keyframes": Object {},
                  "processor": "@delirvfx/posteffect-numeric-slider",
                  "referenceName": null,
                },
                Effect {
                  "expressions": Object {},
                  "id": "layer2",
                  "keyframes": Object {},
                  "processor": "@delirvfx/posteffect-the-world",
                  "referenceName": null,
                },
              ],
              "expressions": Object {},
              "id": "clip-\${idx}",
              "keyframes": Object {},
              "placedFrame": 0,
              "renderer": "video",
            },
            Clip {
              "durationFrames": 0,
              "effects": Array [
                Effect {
                  "expressions": Object {},
                  "id": "layer0",
                  "keyframes": Object {},
                  "processor": "@delirvfx/posteffect-color-slider",
                  "referenceName": null,
                },
                Effect {
                  "expressions": Object {},
                  "id": "layer1",
                  "keyframes": Object {},
                  "processor": "@delirvfx/posteffect-numeric-slider",
                  "referenceName": null,
                },
                Effect {
                  "expressions": Object {},
                  "id": "layer2",
                  "keyframes": Object {},
                  "processor": "@delirvfx/posteffect-the-world",
                  "referenceName": null,
                },
              ],
              "expressions": Object {},
              "id": "clip-\${idx}",
              "keyframes": Object {},
              "placedFrame": 0,
              "renderer": "video",
            },
          ],
          "id": " layerId",
          "name": "mocked-layer",
        },
      ],
      "name": "mocked-composition",
      "samplingRate": 48000,
      "width": 100,
    },
  ],
  "formatVersion": "2017091401",
}
`;
